tests/phpunit/tests/block-template.php

```diff
@@ -226,4 +264,109 @@ public function test_resolve_home_block_template_no_resolution() {
 
 		$this->assertNull( $template );
 	}
+
+	/**
+	 * @ticket 55996
+	 */
+	public function test_get_the_block_template_html_parses_template_content() {
+		global $_wp_current_template_content, $wp_query, $wp_the_query;
+
+		$wp_query     = new WP_Query( array( 'p' => self::$post_id ) );
+		$wp_the_query = $wp_query;
+		the_post();
+
+		$_wp_current_template_content = self::$demo_block_template;
+
+		$html = get_the_block_template_html();
+		$this->assertSame( '<div class="wp-site-blocks">' . self::$demo_parsed_template . '</div>', $html );
+	}
+
+	/**
+	 * @ticket 55996
+	 */
+	public function test_get_the_block_template_html_parses_post_content_only_once() {
+		global $_wp_current_template_content, $wp_query, $wp_the_query;
+
+		// This filter should only be run once per image (and the post here only has a single image).
+		add_filter(
+			'wp_content_img_tag',
+			function( $img ) {
+				return str_replace( '<img ', '<img data-attr="value" ', $img );
+			}
+		);
+
+		$wp_query     = new WP_Query( array( 'p' => self::$post_id ) );
+		$wp_the_query = $wp_query;
+		the_post();
+
+		$_wp_current_template_content = self::$demo_block_template;
+
+		$html = get_the_block_template_html();
+		$this->assertStringContainsString( '<img data-attr="value" ', $html );
+		$this->assertStringNotContainsString( '<img data-attr="value" data-attr="value" ', $html );
```

Comment:
Test methods with multiple assertions should have a final `$message` parameter to explain what went wrong should an assertion fail. Same for other tests in the PR as well.